hydra-node/test/Hydra/HeadLogicSpec.hs

```diff
@@ -274,14 +275,17 @@ spec = do
 -- Assertion utilities
 --
 
-hasEffect :: (HasCallStack, IsTx tx) => Outcome tx -> Effect tx -> IO (HeadState tx)
-hasEffect (NewState s effects) effect
-  | effect `elem` effects = pure s
+hasEffect :: (HasCallStack, IsTx tx) => Outcome tx -> Effect tx -> IO ()
+hasEffect (NewState _ effects) effect
+  | effect `elem` effects = pure ()
+  | otherwise = failure $ "Missing effect " <> show effect <> " in produced effects: " <> show effects
+hasEffect (OnlyEffects effects) effect
+  | effect `elem` effects = pure ()
   | otherwise = failure $ "Missing effect " <> show effect <> " in produced effects: " <> show effects
 hasEffect o _ = failure $ "Unexpected outcome: " <> show o
 
-hasEffect_ :: (HasCallStack, IsTx tx) => Outcome tx -> Effect tx -> IO ()
-hasEffect_ o e = void $ hasEffect o e
+hasEffect_ :: (HasCallStack, IsTx tx) => Outcome tx -> Effect tx -> HeadState tx -> IO (HeadState tx)
+hasEffect_ o e currentState = currentState <$ hasEffect o e
```

Comment:
Should: change the signatures of `hasEffect` and `hasEffect_` back to what they were.

There is a convention in Haskell where a function with `_` in the end, is the same but just without a result. So `hasEffect_` is the same as `hasEffect` but without a return value. Please change it back.